DigitalOffice/Services/UserService/Commands/LoginUserCommand.cs

```diff
@@ -0,0 +1,56 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UserService.Commands.Interfaces;
+using UserService.Database;
+using UserService.Database.Entities;
+using UserService.Models;
+using UserService.Validators;
+using Microsoft.AspNetCore.Cryptography.KeyDerivation;
+using System.Security.Cryptography;
+using System.Text;
+using System.Net;
+using System.Web;
+using UserService.Controllers;
+
+namespace UserService.Commands
+{
+    public class LoginUserCommand : ILoginUserCommand
+    {
+        IValidator<UserCredentials> validator;
+        UserServiceDbContext dbContext;
+        User user;
+
+        SHA512 shaM = new SHA512Managed();
+
+        public LoginUserCommand(
+            [FromServices] IValidator<UserCredentials> validator,
+            [FromServices] UserServiceDbContext dbContext)
+        {
+            this.validator = validator;
+            this.dbContext = dbContext;
+        }
+
+        public Guid Execute(UserCredentials userCredentials)
+        {
+            validator.ValidateAndThrow(userCredentials);
+
+            if((user = dbContext.Users.Where(user => user.Email == userCredentials.Email).FirstOrDefault()) == null)
+            {
+                throw new Exception(); //Has to be changed to NotFound later.
+            }
+
+            if(user.PasswordHash !=
+                Encoding.UTF8.GetString(shaM.ComputeHash(Encoding.UTF8.GetBytes(userCredentials.Password))))
```

Comment:
decompose this code into separate blocks, hard to read

Reply:
fixed